docs/guidelines.md

```diff
@@ -0,0 +1,45 @@
+# Guidelines
+
+## General
+
+### Properties should have public getters and setters
+
+The properties are generated as-is, even if they don't have getter/setter or if getter/setter is not public. Using
+anything else results in C#/contracts mismatch because client types _will_ have public getters and setters.
+
+### Do not use records
```

Comment:
Suggestion: also add a similar note about why `HashSet<T>` and friends are not supported.

Reply:
As with Bartek's suggestion - this will go as an issue for future development.